[Operator] operator crash due to version panic

**Bug description**

The operator for the latest master failed to work because of version panic:
```
# out/linux_amd64/mesh manifest generate
panic: Malformed version: a3703b76cf4745f3d56bf653ed751509be116351-dirty

goroutine 1 [running]:
istio.io/operator/version.init.0()
	istio.io/operator@/version/version.go:46 +0x199
```


[ ] Configuration Infrastructure
[ ] Docs
[ X ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

**Steps to reproduce the bug**

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**

--- a/version/version.go
+++ b/version/version.go
@@ -18,7 +18,6 @@ import (
 	goversion "github.com/hashicorp/go-version"
 
 	pkgversion "istio.io/operator/pkg/version"
-	buildversion "istio.io/pkg/version"
 )
 
 const (
@@ -36,11 +35,6 @@ var (
 func init() {
 	var err error
 	operatorVer := OperatorVersionString
-	// If dockerinfo has a tag (e.g., specified by LDFlags), we will use it as the version of operator
-	tag := buildversion.DockerInfo.Tag
-	if tag != "" && tag != "unknown" {
-		operatorVer = tag
-	}
 	OperatorBinaryGoVersion, err = goversion.NewVersion(operatorVer)
 	if err != nil {
 		panic(err)

